ethereum_history_api/circuits/lib/src/verifiers/receipt.nr

```diff
@@ -12,8 +14,19 @@ pub(crate) fn assert_receipt_equals<LOG_NUM, MAX_LOG_DATA_LEN, MAX_RECEIPT_RLP_L
     assert(false, "Not implemented");
 }
 
-fn assert_tx_idx_equals(key: [u8; MAX_RECEIPT_TREE_DEPTH], tx_idx: Field) {
-    assert(false, "Not implemented");
+pub fn assert_tx_idx_equals(key: [u8; KEY_LEN], tx_idx: Field) {
+    let (key_right_padded, shifted_by_len) = left_nibble_shift(key);
+    let key_bytes: [u8; KEY_LEN / 2] = nibbles_to_bytes(key_right_padded);
+    let (rlp_offset, rlp_len) = decode0(key_bytes);
+    assert(shifted_by_len + 2 * (rlp_len + rlp_offset) == KEY_LEN, "key is not an rlp-encoded string");
+
+    if (rlp_len == 0) {
+        assert(tx_idx == 0);
```

Comment:
Please add an assertion message here & a test for it

Reply:
Added an assertion message and a test. Also, moved the case when rlp_len != 0 to an else, because if rlp_len == 0 we don't need to check all of the stuff below